crates/ty_python_semantic/src/types.rs

```diff
@@ -8294,6 +8310,25 @@ impl<'db> ModuleLiteralType<'db> {
         Some(Type::module_literal(db, importing_file, submodule))
     }
 
+    fn try_module_getattr(self, db: &'db dyn Db, name: &str) -> PlaceAndQualifiers<'db> {
+        // For module literals, we want to try calling the module's own __getattr__ function
+        // if it exists. First, we need to look up the __getattr__ function in the module's scope.
+        if let Some(file) = self.module(db).file(db) {
+            let getattr_symbol = imported_symbol(db, file, "__getattr__", None);
+            if let Place::Type(getattr_type, _) = getattr_symbol.place {
+                // If we found a __getattr__ function, try to call it with the name argument
+                if let Ok(outcome) = getattr_type.try_call(
+                    db,
+                    &CallArguments::positional([Type::string_literal(db, name)]),
+                ) {
+                    return Place::bound(outcome.return_type(db)).into();
```

Comment:
Arguably the boundness here should match the boundness of `__getattr__` itself (which is thrown away using `_` above), but I really don't think this matters in practice.